pkg/api/pod/util.go

```diff
@@ -1373,3 +1379,20 @@ func useOnlyRecursiveSELinuxChangePolicy(oldPodSpec *api.PodSpec) bool {
 	// No feature gate + no value in the old object -> only Recursive is allowed
 	return true
 }
+
+// restartableInitContainersResizePolicy returns true if the pod spec is non-nil and
+// it has any init container with ContainerRestartPolicyAlways and non-nil ResizePolicy.
+func restartableInitContainersResizePolicy(podSpec *api.PodSpec) bool {
+	if podSpec == nil {
+		return false
+	}
+	var resizePolicy bool
+	VisitContainers(podSpec, InitContainers, func(c *api.Container, containerType ContainerType) bool {
+		if IsRestartableInitContainer(c) && len(c.ResizePolicy) > 0 {
+			resizePolicy = true
+			return false
+		}
+		return true
+	})
+	return resizePolicy
```

Comment:
nit: I think iterating directly is simpler here
```suggestion
	for _, c := range podSpec.InitContainers {
		if IsRestartableInitContainer(c) && len(c.ResizePolicy) > 0 {
			return true
		}
	}
	return false
```

Reply:
Made the change.